tests/integration/runtimes/test_runtimes.py

```diff
@@ -0,0 +1,492 @@
+import asyncio
+import json
+import multiprocessing
+import time
+
+import pytest
+
+from jina import Document, Executor, Client, requests
+from jina.enums import PollingType
+from jina.helper import random_port
+from jina.parsers import set_gateway_parser, set_pea_parser
+from jina.peapods.networking import GrpcConnectionPool
+from jina.peapods.runtimes.gateway.grpc import GRPCGatewayRuntime
+from jina.peapods.runtimes.head import HeadRuntime
+from jina.peapods.runtimes.worker import WorkerRuntime
+from jina.types.message.common import ControlMessage
+
+
+@pytest.mark.asyncio
+# test gateway, head and worker runtime by creating them manually in the most simple configuration
+async def test_runtimes_trivial_topology():
+    worker_port = random_port()
+    head_port = random_port()
+    port_expose = random_port()
+    graph_description = '{"start-gateway": ["pod0"], "pod0": ["end-gateway"]}'
+    pod_addresses = f'{{"pod0": ["0.0.0.0:{head_port}"]}}'
```

Comment:
is it a double dict?

Reply:
no thats just escaping the curly bracket in the format string